### PR TITLE
Move to MSBuild 15.0 and Visual Studio 2017

### PR DESCRIPTION
**Pull request by jeffkl**
Move to XUnit

Move to NuGet v4 and PackageReference

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
-    <!-- support building in non msbuild 15 -->
-    <DirectoryBuildPropsImported>true</DirectoryBuildPropsImported>
     <EnlistmentRoot>$(MSBuildThisFileDirectory.TrimEnd('\\'))</EnlistmentRoot>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
 
-    <BaseIntermediateOutputPath Condition=" '$(BaseIntermediateOutputPath)'=='' ">$(EnlistmentRoot)\obj\</BaseIntermediateOutputPath>
-    <IntermediateOutputPath Condition=" '$(IntermediateOutputPath)'=='' ">$(BaseIntermediateOutputPath)$(Configuration)\$(Platform)\$(MSBuildProjectFile)</IntermediateOutputPath>
-    <RestoreOutputPath>$(IntermediateOutputPath)</RestoreOutputPath>
     <OutputPath Condition=" '$(OutputPath)' == '' ">$(EnlistmentRoot)\bin\$(Configuration)\$(Platform)\$(MSBuildProjectName)</OutputPath>
```
Comment (kingerja) [inline Directory.Build.props]: Add the use of the CBT.UnifiedOutput module.

File: src/CBT.Core/CBT.Core.csproj [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <ProjectGuid>{3B843D72-26A8-4A77-B9D2-E90D1C5D521D}</ProjectGuid>
     <OutputType>Library</OutputType>
-    <RootNamespace Condition=" '$(RootNamespace)' == '' ">$(MSBuildProjectName)</RootNamespace>
-    <AssemblyName Condition=" '$(AssemblyName)' == '' ">$(MSBuildProjectName)</AssemblyName>
   </PropertyGroup>
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), build.props))\build.props" />
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" />
```
Comment (kingerja) [inline src/CBT.Core/CBT.Core.csproj]: Shouldn't these be at the top of the project.  We are trying to make it as close to the template.

File: src/CBT.Core.UnitTests/CBT.Core.UnitTests.csproj [hunk 3]
```diff
@@ -1,16 +1,12 @@
...
     <OutputType>Library</OutputType>
-    <RootNamespace Condition=" '$(RootNamespace)' == '' ">$(MSBuildProjectName)</RootNamespace>
-    <AssemblyName Condition=" '$(AssemblyName)' == '' ">$(MSBuildProjectName)</AssemblyName>
     <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
-    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">10.0</VisualStudioVersion>
-    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
     <TestProjectType>UnitTest</TestProjectType>
   </PropertyGroup>
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), build.props))\build.props" />
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" />
```
Comment (kingerja) [inline src/CBT.Core.UnitTests/CBT.Core.UnitTests.csproj]: at top of project like the template would give it to us?

Comment (jeffkl) [inline src/CBT.Core.UnitTests/CBT.Core.UnitTests.csproj]: Yeah I went back and forth.  I think its a better practice to have a `<PropertyGroup />` first and the first `<Import />` after but the templates don't do it that way.  Technically we can do whatever we want with this code base.  ProjectCop would be nice to have here 😄  What do you think, go with template or best practice?

Comment (kingerja) [inline src/CBT.Core.UnitTests/CBT.Core.UnitTests.csproj]: I really like best practice.  However in the sdk projects there is no preceeding property group which means we can't have it as best practice even though it seems insane not to.  

So until we have some ability to change out of box experience with projectcop we want to conform to that experience as best we can to surface real customer issues.

In these projects technically they are ours but we have so few projects that we build cbt against everyone is precious.

**Pull request merged**